src/main/scala/scala/compat/java8/DurationConverters.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2012-2017 Typesafe Inc. <http://www.typesafe.com>
+ */
+package scala.compat.java8
+
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.TimeUnit
+import java.time.{Duration => JavaDuration}
+
+import scala.concurrent.duration.{FiniteDuration, Duration => ScalaDuration}
+
+
+/**
+ * This class contains static methods which convert between Java Durations
+ * and the durations from the Scala concurrency package. This is useful when mediating between Scala and Java
+ * libraries with asynchronous APIs where timeouts for example are often expressed as durations.
+ */
+object DurationConverters {
+
+  /**
+   * Transform a Java duration into a Scala duration. If the nanosecond part of the Java duration is zero the returned
+   * duration will have a time unit of seconds and if there is a nanoseconds part the Scala duration will have a time
+   * unit of nanoseconds.
+   *
+   * @throws IllegalArgumentException If the given Java Duration is out of bounds of what can be expressed with the
+   *                                  Scala Durations.
+   */
+  final def toScala(duration: java.time.Duration): scala.concurrent.duration.Duration = {
+    if (duration.getNano == 0) {
+      if (duration.getSeconds == 0) ScalaDuration.Zero
+      else FiniteDuration(duration.getSeconds, TimeUnit.SECONDS)
+    } else {
+      FiniteDuration(
+        duration.getSeconds * 1000000000 + duration.getNano,
```

Comment:
Should probably be a `Math.multiplyExact` here though and an overflow check after adding the nano part.